packages/opentelemetry-exporter-zipkin/src/zipkin.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import * as http from 'http';
+import * as https from 'https';
+import * as url from 'url';
+import { NoopLogger } from '@opentelemetry/core';
+import {
+  SpanExporter,
+  ReadableSpan,
+  ExportResult,
+} from '@opentelemetry/basic-tracer';
+
+import * as zipkinTypes from './types';
+import { toZipkinSpan } from './transform';
+
+/**
+ * Zipkin Exporter
+ */
+export class ZipkinExporter implements SpanExporter {
+  static readonly DEFAULT_URL = 'http://localhost:9411/api/v2/spans';
+
+  private readonly _url: url.UrlWithStringQuery;
+  private readonly _serviceName: string;
+  private readonly _logger: types.Logger;
+
+  constructor(config: zipkinTypes.ExporterConfig) {
+    const urlStr = config.url || ZipkinExporter.DEFAULT_URL;
+
+    this._url = url.parse(urlStr);
+    this._serviceName = config.serviceName;
+    this._logger = config.logger || new NoopLogger();
+  }
+
+  /**
+   * Export spans.
+   */
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ) {
+    this._logger.debug('Zipkin exporter export');
+    // TODO: buffer spans (batch based on both time and max number)
+    const zipkinSpans = spans.map(this.toZipkinSpan);
+    // TODO: will the caller retry and manage backoff in the case ExportResult
+    // is FailedRetriable?
+    return this.send(zipkinSpans, resultCallback);
+  }
+
+  /**
+   * Shutdown exporter. Noop operation in this exporter.
+   */
+  shutdown() {
+    // TODO: should we initiate an oppurtinistic send(..)?
```

Comment:
This is a good point, I think we should call `send()` one last time to send last batched data.

Reply:
Let's discuss behavior on the next SIG meeting. As we don't have a callback here graceful shutdown is not really possible. I'd love the spec to be more specific here about this behavior: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-exporter.md#shutdown

Reply:
I think it depends on the use case. What is the use case of `shutdown`?